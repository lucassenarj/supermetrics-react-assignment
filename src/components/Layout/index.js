import React from "react";
import Header from "./../Header";
import PropTypes from "prop-types";
import "./index.scss";

function Layout({ children }) {
  return (
    <div className="layout">
      <Header />
      <div>
        { children }
      </div>
    </div>
  )
};

Layout.propTypes = {
  children: PropTypes.element.isRequired,
}

export default Layout;
