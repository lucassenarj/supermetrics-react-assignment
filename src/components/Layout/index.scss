@import "./../../assets/colors.scss";

.layout {
  display: flex;
  flex-direction: column;
  background-color: $backgroundDashboard;
  min-height: 100vh;
}