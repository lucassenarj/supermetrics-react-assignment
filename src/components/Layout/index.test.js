import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Layout from "./";

describe("Components", () => {
  describe("Layout", () => {
    it("renders correctly", () => {
      const { container } = render(
        <MemoryRouter>
          <Layout>
            <h1>Test component</h1>
          </Layout>
        </MemoryRouter>
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
