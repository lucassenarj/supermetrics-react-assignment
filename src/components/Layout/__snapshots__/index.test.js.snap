// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Layout renders correctly 1`] = `
<div
  class="layout"
>
  <header
    class="header"
  >
    <a
      href="/dashboard"
    >
      <img
        alt="Home"
        src="logo.png"
      />
    </a>
    <nav
      class="menu"
    >
      <li
        class="menu-item"
      >
        <a
          href="/dashboard"
        >
          Blog
        </a>
      </li>
      <li
        class="menu-item"
      >
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li
        class="menu-item"
      >
        <a
          href="/contact"
        >
          Contact
        </a>
      </li>
    </nav>
  </header>
  <div>
    <h1>
      Test component
    </h1>
  </div>
</div>
`;
